armi/reactor/blueprints/blockBlueprint.py

```diff
@@ -170,6 +179,46 @@ def construct(
                 runLog.warning(str(e), single=True)
         return b
 
+    def _checkByComponentMaterialInput(self, materialInput):
+        for component in materialInput:
+            if component != "byBlock":
+                if component not in [componentDesign.name for componentDesign in self]:
+                    if materialInput[component]:  # ensure it is not empty
+                        raise ValueError(
+                            f"The component '{component}' used to specify a by-component"
+                            f" material modification is not in block '{self.name}'."
+                        )
+
+    @staticmethod
+    def _filterMaterialInput(materialInput, componentDesign):
```

Comment:
Two thoughts:

1. If this is a `@staticmethod` then we want people to use this method outside this class, so it shouldn't start with an `_`.
2. This method seems cool, and I like it, but it also does a lot of work. Can we have a unit test or three on this method alone?

Reply:
The reason for the `@staticmethod` is just to avoid passing the instance to the method, since the instance is not needed. But I don't expect this method to be used from outside the class, hence the underscore. Per our offline discussion, I am going to leave that unchanged.

Unit test added.